### ISSUE TITLE
A11y: [DateTimePicker] At small, short browser sizes, Time Selection items are obscured

### ISSUE THREAD

**Issue by taleena** [grafana/grafana#64476]
At small, short widths, time selector doesn't always show all information. "Absolute time range" option is obscured. "Search quick ranges" is obscured. 

Low vision users cannot perform the functionality related to the controls which are blocked. Why this size? In addition to users zooming at high levels to see/access, older mobile devices typically have smaller screen sizes, iPad apps in split view (including Safari) can be 320 points wide, desktop users may still resize browser windows to small widths.

**Steps to reproduce**:

- Open https://play.grafana.org/ (or any other) and sign in. "Home" page appears.
- Open chrome dev tools, click on Toggle Device Toolbar in upper left. 
![toggle](https://user-images.githubusercontent.com/27587456/223891656-85244650-b105-4bfd-b565-9387d37f3002.png)
- Use this to set the width and height to 320 x 256
- Click the 3 dots in upper right. Navigate to the icon that looks like clock, a new pane appears. See that "Absolute time range" option is partially obscured. Above it, "Search Quick Ranges" search field is fully obscured. 

**Actual Result**
Obscured:
<img width="388" alt="Screenshot 2023-03-08 at 3 42 45 PM" src="https://user-images.githubusercontent.com/27587456/223889928-aa24e174-a961-41c8-85af-e79aac26413d.png">

**Expected Result**
Behave more like this, where the new pane doesn't extend above the viewport:
<img width="384" alt="Screenshot 2023-03-08 at 3 43 40 PM" src="https://user-images.githubusercontent.com/27587456/223889944-f1ecbbf3-ae45-41d4-9b5d-f66c50bd15e3.png">

**Relevant WCAG Criteria:** [#.#.# WCAG Criterion](link to https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0)
This was tagged with WCAG 2.1 in the document where is was reported, ID 256193

**Environment**:
- Grafana version: 9.4.3
- Data source type & version:
- User OS & Browser: MacOS (latest), Chrome (latest)
- Others:
